### DISCUSSION TITLE
Translation of Adguard script filters doesn't always work

### DISCUSSION THREAD

**Discussion by peace2000** [uBlockOrigin/uBlock-issues#2212] [Interoperability]
I have noticed, that when uBO loads Adguard script filters through a stock list, uBO will translate them into uBO format in asset viewer, even if the rule is originally in Adguard form.

E.g. see this comparison of `FRA: AdGuard Français` in uBO's asset viewer:

![kuva](https://user-images.githubusercontent.com/17256841/184536151-94608a47-026e-464f-9582-0d223a247248.png)

Original format in GH: https://github.com/AdguardTeam/AdguardFilters/blob/master/FrenchFilter/sections/general_extensions.txt

![kuva](https://user-images.githubusercontent.com/17256841/184536183-8e4d6df1-9f32-4365-bac4-584cabe038c7.png)

I have no issues with this behavior, but I was wondering that when I copy an Adguard rule into my custom filters, it's showing up in red as if it's invalid:

![kuva](https://user-images.githubusercontent.com/17256841/184536300-570248c6-24a5-4050-90e2-84e01127cd15.png)

Also, when I load an Adguard list as a custom list https://raw.githubusercontent.com/AdguardTeam/AdguardFilters/master/AnnoyancesFilter/Cookies/sections/cookies_specific.txt and view it through uBO's asset viewer, it's scripts are shown in red as well:

![kuva](https://user-images.githubusercontent.com/17256841/184536444-1b5f283d-a907-4475-bfdd-3e2aff39b7c9.png)

Is it intentional that uBO doesn't load AG scriplet filters that have an uBO equivalent and convert them automatically to uBO filters?

(Yes, I know that uBO doesn't support every AG scriplet filter but I'm speaking of those that it does support).

**Comment by ghost** [comment]
AdGuard has special converter on server side for uBO. Check exactly the link, it will have different ID in URL. (Top right in uBO assets viewer)

**Reply by peace2000** [reply]
Interesting fact. Though why not convert scripts in uBO, just like procedural cosmetics?

**Reply by krystian3w** [reply]
Maybe due similar reason to dropped translate AdGuard HTML filtering syntax.

https://kb.adguard.com/en/general/how-to-create-your-own-ad-filters#html-filtering-rules
https://github.com/AdguardTeam/tsurlfilter/issues/55#issuecomment-1151886541

**Reply by krystian3w** [reply]
Invalid translation scriptlets maybe report here: https://github.com/AdguardTeam/Scriptlets/issues/237

**Reply by Yuki2718** [reply]
Please report if you find any imcompatibility. Translation is done just automatically and thus not always valid:
https://github.com/AdguardTeam/Scriptlets/issues/119
https://github.com/AdguardTeam/Scriptlets/issues/120

**Reply by peace2000** [reply]
What I meant was that is it intentional that uBO doesn't convert AG scripts to uBO format if AG scripts are found in custom filters / lists.

**Reply by Yuki2718** [reply]
IDK, $$ was rejected as it requires too much code change but not sure about #%#. In any case I don't see too much problem as all the AG lists have uBO version.

**Reply by krystian3w** [reply]
AG server for ra and rc should be converted with stay / asap stay parameter due AdGuard like "loop" after Alex 302 issue to add mutation observer as default state.

Maybe this should be checked and reported if "slavalenka" don't added injection parameter.

**Reply by Yuki2718** [reply]
@krystian3w You mean they're not correctly converted? Example?

**Reply by krystian3w** [reply]
E.g. default behaviour in AG is check reinjection class or attribs, but in uBo nope; so to correct work these arsenal in uBo translation file needed is add stay or asap stay to better compatibility with idea used in AG environment.

**Reply by Yuki2718** [reply]
Yeah, the default behaviror of AG's ra/rc is almost those with stay on uBO. Maybe you can open an issue.

**Reply by krystian3w** [reply]
Good luck: https://github.com/AdguardTeam/Scriptlets/issues/245
